src/plone/api/env.py

```diff
@@ -201,6 +202,17 @@ def test_mode():
     return IS_TEST
 
 
+def read_only_mode():
+    """Returns True if you are running the zope instance on an read only ZODB.
+
+    :Example: :ref:`env_read_only_mode_example`
+    """
+    conn = Globals.DB.open()
```

Comment:
You will need just a connection, it does not matter if it is a new or a shared one.

If you have code for getting the current or one from the pool, your are welcome to improve it.